internal/pkg/jsonnet/context_test.go

```diff
@@ -0,0 +1,111 @@
+package jsonnet
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-jsonnet"
+	"github.com/google/go-jsonnet/ast"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVmContext_Nil(t *testing.T) {
+	t.Parallel()
+	output, err := Evaluate(`{foo: "bar"}`, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, "{\n  \"foo\": \"bar\"\n}\n", output)
+}
+
+func TestVmContext_Empty(t *testing.T) {
+	t.Parallel()
+	ctx := NewContext()
+	output, err := Evaluate(`{foo: "bar"}`, ctx)
+	assert.NoError(t, err)
+	assert.Equal(t, "{\n  \"foo\": \"bar\"\n}\n", output)
+}
+
+func TestVmContext_Complex(t *testing.T) {
+	t.Parallel()
+	ctx := NewContext()
+	ctx.ExtVar("var1", "value1")
+	ctx.ExtVar("var2", "value2")
+	ctx.NativeFunction(&jsonnet.NativeFunction{
+		Name:   `func1`,
+		Params: ast.Identifiers{"param1", "param2"},
+		Func: func(params []interface{}) (interface{}, error) {
+			return fmt.Sprintf("---%s---%s---", params[0], params[1]), nil
+		},
+	})
+	ctx.NativeFunctionWithAlias(&jsonnet.NativeFunction{
+		Name:   `func2`,
+		Params: ast.Identifiers{"param1", "param2"},
+		Func: func(params []interface{}) (interface{}, error) {
+			return fmt.Sprintf("***%s***%s***", params[0], params[1]), nil
+		},
+	})
+	ctx.LocalAlias(`alias1`, `"aliasValue"`)
+	ctx.LocalAlias(`alias2`, `1+2`)
+
+	code := `
+{
+  var1: std.extVar("var1"),
+  var2: std.extVar("var2"),
+  func1: std.native("func1")("param1", "param2"),
+  func2: std.native("func2")("param1", "param2"),
+  alias1: alias1,
+  alias2: alias2,
+  func2Alias: func2("param1", "param2"),
+}
+`
+
+	expected := `
+{
+  "alias1": "aliasValue",
+  "alias2": 3,
+  "func1": "---param1---param2---",
+  "func2": "***param1***param2***",
+  "func2Alias": "***param1***param2***",
+  "var1": "value1",
+  "var2": "value2"
+}
```

Comment:
Pozri prosim najprv tento test, z toho by to malo byt jasne.

Narazil som na to, ze budeme potrebovat definovat aj vlastne JsonNet funkcie a aliases.
Tak som nahradil `jsonnet.VariablesValues` -> `jsonnet.Context`.

Vsetko je to otestovane v tomto test:
- ext variable
- native (Go) function
- native (Go) function + alias (globalna premenna)
- alias (globalna premenna)

Reply:
Note: ako vidis, JsonNet kniznica zoradzuje kluce podla abecedy, zatial neriesime.